packages/wonder-blocks-tooltip/components/tooltip-anchor.test.js

```diff
@@ -0,0 +1,488 @@
+        };
+
+        const actAndAssert = (ref) => {
+            // Act
+            // Let's fake a focusin (this is the event that the anchor gets
+            // whether focused directly or a child is focused). We have to
+            // fake directly because there's no real browser here handling
+            // focus and real events.
+            const act = doEvent(ref, new FocusEvent("focusin"));
+
+            // Need a timeout here so that the event handling can occur.
+            act.then(() => {
+                const result = ref && ref.innerHTML;
+
+                // Assert
+                expect(result).toBe("true");
+                done();
+            });
+        };
+
+        arrange(actAndAssert);
+    });
+
+    describe("loses keyboard focus", () => {
+        test("active is set to false", (done) => {
+            // Arrange
+            const arrange = (actAndAssert) => {
+                const getFakeTooltipPortalMounter = (active) =>
+                    (((
+                        <View id="anchor">{active ? "true" : "false"}</View>
+                    ): any): React.Element<typeof TooltipPortalMounter>);
+                const nodes = (
+                    <View>
+                        <TooltipAnchor anchorRef={actAndAssert}>
+                            {getFakeTooltipPortalMounter}
+                        </TooltipAnchor>
+                    </View>
+                );
+                mount(nodes);
+            };
+
+            const actAndAssert = (ref) => {
+                // Act
+                const act = doEvent(ref, new FocusEvent("focusin")).then(() =>
+                    doEvent(ref, new FocusEvent("focusout")),
+                );
+
+                // Assert
+                act.then(() => {
+                    const result = ref && ref.innerHTML;
+                    expect(result).toBe("false");
+                    done();
+                });
+            };
+
+            arrange(actAndAssert);
+        });
+
+        test("if hovered, remains active", (done) => {
+            // Arrange
+            const arrange = (actAndAssert) => {
+                const getFakeTooltipPortalMounter = (active) =>
+                    (((
+                        <View id="anchor">{active ? "true" : "false"}</View>
+                    ): any): React.Element<typeof TooltipPortalMounter>);
+                const nodes = (
+                    <View>
+                        <TooltipAnchor anchorRef={actAndAssert}>
+                            {getFakeTooltipPortalMounter}
+                        </TooltipAnchor>
+                    </View>
+                );
+                mount(nodes);
+            };
+
+            const actAndAssert = (ref) => {
+                // Act
+                const act = doEvent(ref, new FocusEvent("focusin"))
+                    .then(() => doEvent(ref, new MouseEvent("mouseenter")))
+                    .then(() => doEvent(ref, new FocusEvent("focusout")));
```

Comment:
This is better.  I was actually thinking:
```
test("if hovered, remains active", async () => {
   ...
   await doEvent(ref, new FocusEvent("focusin"));
   await doEvent(ref, new MouseEvent("mouseenter"));
   await doEvent(ref, new FocusEvent("focusout"));
   const result = ref & ref.innerHTML;
   expect(result).toBe("true");
});
```
See http://jestjs.io/docs/en/tutorial-async.html#async-await.  Let's leave it as a TODO for now.  I think using `async`/`await` could simplify some of the other tests here as well.

Reply:
Oh yeah, I forgot about that syntax getting added. Used to it from C#.